test/test_gen_backend_stubs.py

```diff
@@ -0,0 +1,100 @@
+class TestGenBackendStubs(TestCase):
+
+    def get_errors_from_gen_backend_stubs(self, yaml_str: str) -> str:
+        with tempfile.NamedTemporaryFile(mode='w') as fp:
+            fp.write(yaml_str)
+            fp.flush()
+            try:
+                tools.codegen.gen_backend_stubs.run(fp.name, '', True)
+            except AssertionError as e:
+                # Scrub out the temp file name from any error messages to simplify assertions.
+                return str(e).replace(fp.name, '')
+            self.fail('Expected gen_backend_stubs to raise an AssertionError, but it did not.')
+
+    def test_missing_backend(self):
+        yaml_str = '''\
+cpp_namespace: torch_xla
+supported:
+- abs'''
+        output_error = self.get_errors_from_gen_backend_stubs(yaml_str)
+        self.assertExpectedInline(output_error, '''You must provide a value for "backend"''')
+
+    def test_empty_backend(self):
+        yaml_str = '''\
+backend:
+cpp_namespace: torch_xla
+supported:
+- abs'''
+        output_error = self.get_errors_from_gen_backend_stubs(yaml_str)
+        self.assertExpectedInline(output_error, '''You must provide a value for "backend"''')
+
+    def test_missing_cpp_namespace(self):
+        yaml_str = '''\
+backend: XLA
+supported:
+- abs'''
+        output_error = self.get_errors_from_gen_backend_stubs(yaml_str)
+        self.assertExpectedInline(output_error, '''You must provide a value for "cpp_namespace"''')
+
+    def test_whitespace_cpp_namespace(self):
+        yaml_str = '''\
+backend: XLA
+cpp_namespace:\t
+supported:
+- abs'''
+        output_error = self.get_errors_from_gen_backend_stubs(yaml_str)
+        self.assertExpectedInline(output_error, '''You must provide a value for "cpp_namespace"''')
+
+    def test_empty_supported(self):
+        yaml_str = '''\
+backend: XLA
+cpp_namespace: torch_xla
+supported:
+'''
+        output_error = self.get_errors_from_gen_backend_stubs(yaml_str)
+        self.assertExpectedInline(output_error, '''expected "supported" to be a list, but got: None (of type <class 'NoneType'>)''')
+
+    # supported is a single item (it should be a list)
+    def test_nonlist_supported(self):
+        yaml_str = '''\
+backend: XLA
+cpp_namespace: torch_xla
+supported: abs'''
+        output_error = self.get_errors_from_gen_backend_stubs(yaml_str)
+        self.assertExpectedInline(output_error, '''expected "supported" to be a list, but got: abs (of type <class 'str'>)''')
+
+    # supported contains an op that isn't in native_functions.yaml
+    def test_supported_invalid_op(self):
+        yaml_str = '''\
+backend: XLA
+cpp_namespace: torch_xla
+supported:
+- abs_BAD'''
+        output_error = self.get_errors_from_gen_backend_stubs(yaml_str)
+        self.assertExpectedInline(output_error, '''Found an invalid operator name: abs_BAD. For the list of valid operator names, see https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml''')
```

Comment:
`# noqa: E501`, I think. (I used to just `# noqa` these but it looks like @samestep doesn't like that in #56272). Actually, I noticed people have been manually adding line breaks to the expect tests in `test/test_dispatch.py` which is bad, because it means the automatic update mechanism will no longer work.

Reply:
For long lines I think it's `# noqa: B950` (at least in the PyTorch codebase, maybe it's E501 elsewhere)